src/lib/actionGenerator.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * @file actionGenerator.js
+ * Exports a generator that assists in the construction of async saga actions.
+ */
+
+import { put } from 'redux-saga/effects';
+import {
+  showLoading,
+  hideLoading,
+  resetLoading
+} from 'react-redux-loading-bar';
+
+/**
+ * Generator that handles loading/error states.
+ *
+ * @param {string} type - This action's machine-friendly name.
+ * @param {function} actionHandler - Generator function containing loaders.
+ *
+ * @returns {undefined} nothing.
+ */
+export default function* actionGenerator(type, actionHandler) {
```

Comment:
Alright so this is the actionGenerator I was talking about earlier in my comment thread. It basically makes it easy for us to consistently use the loading bar, and yield LOADING, and FAIL states when async calls fail.

Reply:
ya, this is cool. nicely done